assets/src/edit-story/components/library/panes/media/mediaPane.js

```diff
@@ -182,69 +222,97 @@ function MediaPane(props) {
     )
     .map((attachment) => getResourceFromAttachment(attachment));
 
+  const refContainer = useRef();
+  const refContainerFooter = useRef();
+
+  useIntersectionEffect(
+    refContainerFooter,
+    (entry) => {
+      if (!isMediaLoaded || isMediaLoading) return;
+      if (!hasMore) return;
+      if (!entry.isIntersecting) return;
+
+      setPage({ page: page + 1 });
+    },
+    {
+      root: refContainer.current,
```

Comment:
This will start as a `null`. Probably not what you want?

Reply:
I check it [here](https://github.com/google/web-stories-wp/pull/562/files/96e2fb4ba24a03d802a4f7a877506afed9083506#diff-9e607ff4ff88856913ed42eb4b275859R29) (`node` and `root` if you are using that option)

Reply:
That's not enough. The ref starts as null and only set post rendering. I think you should pass `ref` here. Otherwise, by the time you reach [here](https://github.com/google/web-stories-wp/pull/562/files/96e2fb4ba24a03d802a4f7a877506afed9083506#diff-9e607ff4ff88856913ed42eb4b275859R29), the `options.root` is `null`. It's the same reason why you pass the `refContainerFooter` as a ref.

Reply:
Right, so I did it in the right way, tested and then broke it with quick "fix"  -> https://github.com/google/web-stories-wp/pull/562#discussion_r392902349

CC @miina so we both learn 😄

Reply:
Yeah. That happens a lot, but very very dangerous since the re-rendering is involved. If you want to use the node directly, you will need to include it in the `deps` array. But I think `ref` is a cleaner approach here. Also certainly it messes up a bunch of existing APIs from time to time.